tools/toolchains/iar.py

```diff
@@ -126,17 +126,22 @@ def cc_extra(self, object):
         base, _ = splitext(object)
         return ["-l", base + '.s.txt']
 
-    def get_compile_options(self, defines, includes):
+    def get_compile_options(self, defines, includes, for_asm=False):
         opts = ['-D%s' % d for d in defines] + ['-f', self.get_inc_file(includes)]
-        config_header = self.get_config_header()
-        if config_header is not None:
-            opts = opts + ['--preinclude', config_header]
+        if for_asm:
+            # The assembler doesn't support '--preinclude', so we need to add
+            # the macros directly
+            opts = opts + ['-D%s' % d for d in self.get_config_macros()]
```

Comment:
@bogdanm This won't work with values that contain {} and "'. But in general I think we should filter these as assembler can't use strings anyway. E.g. this could be a filtered list of the macros that are defined and/or MACRO=<int> statement. What do you think?